odoo/orm/registry.py

```diff
@@ -160,19 +162,26 @@ def new(
             registry.setup_signaling()
             # This should be a method on Registry
             from odoo.modules.loading import load_modules, reset_modules_state  # noqa: PLC0415
+            exit_stack = ExitStack()
```

Comment:
You seem to be afraid of introducing a `with` statementh here.  Why?

Reply:
Could do it, but had to indent all and people like minimal diffs. ExitStack does nothing when entering and we have a try-finally already here.